core/src/main/java/io/grpc/internal/ProxySocketAddress.java

```diff
@@ -18,26 +18,25 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
-import io.grpc.Attributes;
 import java.net.SocketAddress;
 
 /**
- * A data structure to associate a {@link SocketAddress} with {@link Attributes}.
+ * A data structure to associate a {@link SocketAddress} with {@link ProxyParameters}.
  */
-final class PairSocketAddress extends SocketAddress {
+final class ProxySocketAddress extends SocketAddress {
```

Comment:
We are going to want/need this to be public. That's fine not to include in this PR.

Reply:
Are you thinking public in `io.grpc.internal` or public in `io.grpc`?

Reply:
Public in `io.grpc`. External `NameResolver`s need to be able to configure proxy usage.